source/Files.cpp

```diff
@@ -479,6 +489,111 @@ void Files::Delete(const string &filePath)
 
 
 
+int Files::DeleteDir(const string &path)
+{
+#if defined (_WIN32)
+	HANDLE hFile; // Handle to directory
+	std::string strFilePath; // Filepath
+	std::string strPattern = path + "\\*.*"; // Pattern
+	WIN32_FIND_DATAW FileInformation; // File information
+
+	hFile = FindFirstFileW(Utf8::ToUTF16(strPattern).c_str(), &FileInformation);
+	if(hFile != INVALID_HANDLE_VALUE)
+	{
+		do {
+			if(FileInformation.cFileName[0] != '.')
+			{
+				strFilePath.erase();
+				strFilePath = path + "\\" + Utf8::ToUTF8(FileInformation.cFileName);
+
+				if(FileInformation.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
+				{
+					// Delete subdirectory
+					int iRC = DeleteDir(strFilePath);
+					if(iRC)
+						return iRC;
+				}
+				else
+				{
+					// Set file attributes
+					if(SetFileAttributesW(Utf8::ToUTF16(strFilePath).c_str(), FILE_ATTRIBUTE_NORMAL) == FALSE)
+						return GetLastError();
+
+					// Delete file
+					if(DeleteFileW(Utf8::ToUTF16(strFilePath).c_str()) == FALSE)
+						return GetLastError();
+				}
+			}
+		} while(FindNextFileW(hFile, &FileInformation) == TRUE);
```

Comment:
cant a simple while work for this?

Reply:
Sure we could but I think this is way more elegant